castai/resource_castai_cluster.go

```diff
@@ -138,8 +267,16 @@ func resourceCastaiClusterCreateOrUpdate(ctx context.Context, data *schema.Resou
 	if err != nil {
 		return diag.FromErr(err)
 	}
-
 	log.Printf("[INFO] Cluster %q has reached `ready` status", data.Id())
+
+	log.Printf("[DEBUG] Cluster %q autoscaling policies update", data.Id())
```

Comment:
I think this method `resourceCastaiClusterCreateOrUpdate` is not handling update correctly as it always tries to create new cluster. Maybe it would make more sense to split this into two separate methods as cluster creation and update is quite different.